R/relu.R

```diff
@@ -0,0 +1,144 @@
+#'   If \code{reverse} is true, then the transformation is reflected about the
+#'   y-axis, like so: \deqn{max(0, c - x)} Setting the \code{smooth} option
+#'   to true will instead calculate a smooth approximation to ReLu
+#'   according to \deqn{ln(1 + e^(x - c)} The \code{reverse} argument may
+#'   also be applied to this transformation.
+#'
+#' @section Connection to MARS:
+#'
+#' The rectified linear transformation is used in the Multivariate Adaptive
+#' Regression Splines as basis function to fit piecewise linear functions to
+#' data in a strategy similar to that employeed in tree based models. The
+#' transformation is a population choice as an activation function in many
+#' neural networks, which could then be seen as a stacked generalization of
+#' MARS when making use of ReLu activations. The hinge function also appears
+#' in the loss function of Support Vector Machines, where it penalizes
+#' residuals only if they are within a certain margin of the decision boundary.
+#'
+#' @examples
+#' data(biomass)
+#'
+#' biomass_tr <- biomass[biomass$dataset == "Training",]
+#' biomass_te <- biomass[biomass$dataset == "Testing",]
+#'
+#' rec <- recipe(HHV ~ carbon + hydrogen + oxygen + nitrogen + sulfur,
+#'               data = biomass_tr)
+#'
+#' transformed_te <- rec %>%
+#'   step_relu(carbon, shift = 40) %>%
+#'   prepare(biomass_tr) %>%
+#'   bake(biomass_te)
+#'
+#' transformed_te
+#'
+#' @seealso \code{\link{recipe}} \code{\link{prepare.recipe}}
+#'   \code{\link{bake.recipe}}
+step_relu <-
+  function(recipe,
+           ...,
+           role = "predictor",
+           trained = FALSE,
+           shift = 0,
+           reverse = FALSE,
+           smooth = FALSE,
+           prefix = "right_relu_") {
+    if (!is.numeric(shift))
+      stop("Shift argument must be a numeric value.", call. = FALSE)
+    if (!is.logical(reverse))
+      stop("Reverse argument must be a logical value.", call. = FALSE)
+    if (!is.logical(smooth))
+      stop("Smooth argument must be logical value.", call. = FALSE)
+    if (reverse & prefix == "right_relu_")
+      prefix <- "left_relu_"
+    add_step(
+      recipe,
+      step_relu_new(
+        terms = check_ellipses(...),
+        role = role,
+        trained = trained,
+        shift = shift,
+        reverse = reverse,
+        smooth = smooth,
+        prefix = prefix
+      )
+    )
+  }
+
+step_relu_new <-
+  function(terms = NULL,
+           role = NA,
+           trained = FALSE,
+           shift = 0,
+           reverse = FALSE,
+           smooth = FALSE,
+           prefix = "right_relu_") {
+    step(
+      subclass = "relu",
+      terms = terms,
+      role = role,
+      trained = trained,
+      shift = shift,
+      reverse = reverse,
+      smooth = smooth,
+      prefix = prefix
+    )
+  }
+
+prepare.step_relu <- function(x, training, info = NULL, ...) {
+  x$trained <- TRUE
+  x
+}
+
+#' @importFrom dplyr select mutate_all select_vars tbl_vars funs
```

Comment:
No need to import these functions if you are calling them with a namespace-qualifier as you do below